src/uniprotkb/components/protein-data-views/GoCamViz.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+import { Loader } from 'franklin-sites';
+import * as goCamVizLoader from '@geneontology/wc-gocam-viz/loader';
+
+import styles from './styles/go-cam-viz.module.scss';
+
+type Props = {
+  id: string;
+};
+
+const GoCamViz = ({ id }: Props) => {
+  const [defined, setDefined] = useState(false);
+  useEffect(() => {
+    goCamVizLoader.defineCustomElements();
+    customElements.whenDefined('wc-gocam-viz').then(() => {
+      setDefined(true);
```

Comment:
Async state-setting logic within a useEffect is something to watch out for. Their could be the case of the promise resolving after the component has unmounted, and then it'll try to set a state for a component that doesn't exist anymore.
Maybe use `useSafeState` instead of `useState`.
Have a look at the logic in `useCustomElement`, which internally uses `useSafeState` to do similar things that you're doing here. `useSafeState` is basically a wrapper to make sure that there is no set state when the component is unmounted.